### PR TITLE
add better failure to dbus-wait

### PR DESCRIPTION
**Pull request by shawaj**
**Why**

add better failure

**How**

add extra test

**References**

Closes: #37

### CODE REVIEW

File: dbus-wait.sh [hunk 1]
```diff
@@ -8,7 +8,10 @@ wait_for_dbus() {
 			  /org/freedesktop/DBus \
 			  org.freedesktop.DBus.ListNames
 
-		if [ $? ]; then
```
Comment (marvinmarnold) [inline dbus-wait.sh:11]: Here's an example that shows why I think this did not work:

```
➜  cat fail.sh
false
if [ $? ]; then
	echo "LAST TREATED AS SUCCESS: But it failed"
else
	echo "LAST TREATED AS FAILURE: That's right"
fi

➜  sh ./fail.sh 
LAST TREATED AS SUCCESS: But it failed
➜  cat work.sh 
false
if [ $? -ne 0 ]; then
	echo "LAST TREATED AS FAILURE: That's right"
else
	echo "LAST TREATED AS SUCCESS: But it failed"
fi
➜  sh ./work.sh 
LAST TREATED AS FAILURE: That's right
```

File: dbus-wait.sh [hunk 2]
```diff
@@ -8,7 +8,10 @@ wait_for_dbus() {
...
+		dbus_test="$?"
```
Comment (marvinmarnold) [inline dbus-wait.sh:11]: After some reading, I found [this StackOverflow](https://stackoverflow.com/questions/26675681/how-to-check-the-exit-status-using-an-if-statement) to be most helpful. In this case, it would look like:

```
dbus_test=$?
if [ $dbus_test -ne 0 ]; then
```

File: dbus-wait.sh [hunk 3]
```diff
@@ -8,7 +8,10 @@ wait_for_dbus() {
...
+                if $dbus_test | grep -q 'Failed to connect to socket'; then
```
Comment (marvinmarnold) [inline dbus-wait.sh:12]: Do you think that `dbus-send` will sometimes return a 0 exit code when the message contains 'Failed to connect to socket'?

Also not clear if you are trying to pipe (`|`) or use or (`||`).

File: dbus-wait.sh (hunk 2 above)
Comment (shawaj) [reply dbus-wait.sh:11]: Basically the issue was that when showing "failed to connect to socket" it was still proceeding using the existing code, as shown in the linked comment above:

```
08.10.21 22:01:00 (+0200) helium-miner Failed to open connection to "system" message bus: Failed to connect to socket /host/run/dbus/session_bus_socket: No such file or directory
08.10.21 22:01:00 (+0200) helium-miner DBus is now accepting connections
```

So the idea was to have it so that if it says "can't connect to socket" or returns no output, then it will sleep. Otherwise it will continue.

File: dbus-wait.sh (hunk 3 above)
Comment (shawaj) [reply dbus-wait.sh:12]: Was trying to pipe to grep yes. To check for the string.

Basically the issue was that when showing "failed to connect to socket" it was still proceeding using the existing code, as shown in the linked comment above:

```
08.10.21 22:01:00 (+0200) helium-miner Failed to open connection to "system" message bus: Failed to connect to socket /host/run/dbus/session_bus_socket: No such file or directory
08.10.21 22:01:00 (+0200) helium-miner DBus is now accepting connections
```

So the idea was to have it so that if it says "can't connect to socket" or returns no output, then it will sleep. Otherwise it will continue.

File: dbus-wait.sh (hunk 1 above)
Comment (shawaj) [reply dbus-wait.sh:11]: Aha so adding -ne 0 should fix it?

File: dbus-wait.sh (hunk 3 above)
Comment (marvinmarnold) [reply dbus-wait.sh:12]: tl;dr: ya I think adding `-ne 0` is sufficient.

My understanding is that "Failed to open connection..." is just a message that is outputted to [stderr](https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12/tools/dbus-send.c#L406) during program execution. This is an orthogonal concern to the [numeric exit code](https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html) returned by every linux process, where only 0 marks success. This error message corresponds to an [error code of 1](https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12/tools/dbus-send.c#L411).

The previous code was always evaluating to true. `if [ $? ]` is the same as ` if test $?`, which always evaluated to `if test 1` or `if test 2`. Both of those will always evaluate to true. Here's how I proved it to myself.

```
➜  test 1 -gt 2 && echo "True" || echo "False"
False
➜  test 0 && echo "True" || echo "False"
True
➜  test 1 && echo "True" || echo "False"
True
```

Comment (marvinmarnold) [reply dbus-wait.sh:12]: Clarification: The way the code was organized _originally_, we actually want. `if [ $? -eq 0]`. 

```
                if [ $? -eq 0 ]; then
			break;
		else
			sleep 0.1
```
